crates/worker/src/partition/state_machine/entries/notification.rs

```diff
@@ -7,37 +7,151 @@
+                    tags = (rpc.service = invocation_metadata
+                        .invocation_target
+                        .service_name()
+                        .to_string()),
+                    fields = (with_start_time = header.append_time)
+                );
+            }
+            _ => unreachable!("Invalid matches! filter"),
+        }
+
+        Ok(())
+    }
+}
+
 impl<'e, 'ctx: 'e, 's: 'ctx, S> CommandHandler<&'ctx mut StateMachineApplyContext<'s, S>>
     for ApplyNotificationCommand<'e>
 where
     S: ReadOnlyJournalTable,
 {
     async fn apply(self, ctx: &'ctx mut StateMachineApplyContext<'s, S>) -> Result<(), Error> {
-        if cfg!(debug_assertions)
-            && let NotificationId::CompletionId(completion_id) = self.entry.id()
-        {
-            assert!(
-                ctx.storage
-                    .get_command_by_completion_id(self.invocation_id, completion_id)
-                    .await?
-                    .is_some(),
-                "For given completion id {completion_id}, the corresponding command must be present already in the journal"
-            )
+        if ctx.is_leader {
+            let invocation_metadata = self
+                .invocation_status
+                .get_invocation_metadata()
+                .expect("In-Flight invocation metadata must be present");
+
+            match self.entry.id() {
+                NotificationId::CompletionId(completion_id) => {
+                    self.create_trace_span(ctx, completion_id, invocation_metadata)
+                        .await?;
+                }
+                NotificationId::SignalIndex(index) => {
+                    let _span = instrumentation::info_invocation_span!(
+                        relation = invocation_metadata
+                            .journal_metadata
+                            .span_context
+                            .as_parent(),
+                        id = self.invocation_id,
+                        name = format!("awakeable ({})", index),
+                        tags = (rpc.service = invocation_metadata
+                            .invocation_target
+                            .service_name()
+                            .to_string())
+                    );
+                }
+                NotificationId::SignalName(name) => {
+                    let _span = instrumentation::info_invocation_span!(
+                        relation = invocation_metadata
+                            .journal_metadata
+                            .span_context
+                            .as_parent(),
+                        id = self.invocation_id,
+                        name = format!("awakeable ({})", name),
+                        tags = (rpc.service = invocation_metadata
+                            .invocation_target
+                            .service_name()
+                            .to_string())
+                    );
+                }
```

Comment:
For now remove this, and just leave a comment with a todo for me :) when we get to expose this, i'll deal with it (i don't think the name was supposed to  be awakeable)